protocol.md

```diff
@@ -191,36 +191,60 @@ N  END_SYSEX                (0xF7)
 Capability Query
 ---
 
-The capability query provides a list of all modes supported by each pin and
-the resolution used by each pin. Each pin has 2 bytes for each supported mode
-and a value of 127 to mark the end of that pin's data. The number of pins
-supported is inferred by the message length. The resolution information may be
-used to adapt to future implementation where PWM, analog input and others may
-have different values (such as 12 or 14 bit analog instead of 10-bit analog).
-*For some features such as i2c, the resolution information is less important so
-a value of 1 is used.*
+The capability query provides a list of all modes supported by each pin and the resolution used by each pin. Each pin has 2 bytes for each supported mode and a value of 127 to mark the end of that pin's data. The number of pins supported is inferred by the message length.
 
-Capabilities query
-```
+### Capabilities query
+```c++
 0  START_SYSEX              (0xF0)
-1  capabilities query       (0x6B)
+1  CAPABILITY_QUERY         (0x6B)
 2  END_SYSEX                (0xF7)
 ```
 
-Capabilities response
-```
+### Capabilities response
+```c++
 0  START_SYSEX              (0xF0)
-1  capabilities response    (0x6C)
-2  1st mode supported of pin 0
+1  CAPABILITY_RESPONSE      (0x6C)
+2  1st supported mode of pin 0
 3  1st mode's resolution of pin 0
-4  2nd mode supported of pin 0
+4  2nd supported mode of pin 0
 5  2nd mode's resolution of pin 0
-... additional modes/resolutions, followed by a single 127 to mark the end of
-    the pin's modes. Each pin follows with its mode and 127 until all pins are
-    implemented.
-N  END_SYSEX                 (0xF7)
+... additional modes/resolutions, followed by PIN_MODE_IGNORE,
+    to mark the end of the pin's modes. Subsequently, each pin
+    follows with its modes/resolutions and a PIN_MODE_IGNORE,
+    until all pins are implemented.
+N  END_SYSEX                (0xF7)
+```
+
+#### Supported Modes
+Each supported mode of operation on a pin is identified by one of the following constants found in FirmataConstants.h:
```

Comment:
I wouldn't reference FirmataConstants.h here since I'd like the protocol to remain abstract from any particular implementation. I do realize I need to do some cleanup in the protocol documentation regarding this point. It's okay however to reference a specific implementation to use as an example.